pkg/clusterdomains/controllers/clusterdomainstatus.go

```diff
@@ -30,6 +32,10 @@ const (
 	createCdsRetryInterval               = 10 * time.Second
 )
 
+var (
+	clusterIDRegex = regexp.MustCompile("[^a-zA-Z0-9-.]+")
```

Comment:
Make this a const?

Reply:
You can't assign regexp.MustCompile to a constant.
Assigning it to a global variable, means we only compile the regex once.